Classes/Service/WebAuthnAuthenticationService.php

```diff
@@ -40,31 +46,64 @@ class WebAuthnAuthenticationService extends AuthenticationService
      */
     private $webAuthnSession;
 
+    /**
+     * @var array
+     */
+    private $backendExtensionConfiguration;
+
     public function __construct()
     {
         $this->backendUserRepository = GeneralUtility::makeInstance(ObjectManager::class)
             ->get(BackendUserRepository::class);
         $this->webAuthnService = WebAuthnServiceFactory::fromGlobals();
         $this->webAuthnSession = GeneralUtility::makeInstance(WebAuthnSession::class);
+        $this->backendExtensionConfiguration = GeneralUtility::makeInstance(ExtensionConfiguration::class)->get('cvc_webauthn');
+    }
+
+    public function processLoginData(array &$loginData, $passwordTransmissionStrategy)
+    {
+        $isProcessed = false;
+        if ($passwordTransmissionStrategy === 'normal') {
+            $loginData['webauthn-uident'] = GeneralUtility::_GP('webauthn-userident');
+            if ($loginData['uident'] == '') {
+                $loginData['uident'] = $loginData['webauthn-uident'];
+            }
+            $isProcessed = true;
+        }
+
+        return $isProcessed;
     }
 
     /**
      * @throws \TYPO3\CMS\Core\Crypto\PasswordHashing\InvalidPasswordHashException
      */
     public function authUser(array $user): int
     {
-        $username = (string) $user['username'];
-        $data = base64_decode($this->login['uident']);
+        if ($this->backendExtensionConfiguration['secondFactorLogin'] == '1') {
+            $this->serviceChainValue = parent::authUser($user);
 
-        $beUser = $this->backendUserRepository->findOneByUserName($username);
+            if ($this->serviceChainValue == 200) {
+                $this->serviceChainValue = $this->authenticateUser((string) $user['username']);
+            }
+        } else {
+            $this->serviceChainValue = $this->authenticateUser((string) $user['username']);
+        }
+
+        return $this->serviceChainValue;
+    }
 
+    private function authenticateUser($username): int
+    {
+        $serviceChainValue = 0;
+        $data = base64_decode($this->login['webauthn-uident']);
+        $beUser = $this->backendUserRepository->findOneByUserName($username);
         //if backendUser does not exist authentication is failed
         if ($beUser === null) {
-            return 0;
+            $serviceChainValue = 0;
```

Comment:
The variable is unused, because the value gets overwritten immediately in line `118` or in line `121`.

Reply:
fixed with new commit